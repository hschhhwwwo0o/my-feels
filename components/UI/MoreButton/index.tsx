import React, { FunctionComponent } from "react";

interface IMoreButton {
  onClickHandler: () => any;
}

const MoreButton: FunctionComponent<IMoreButton> = ({ onClickHandler = () => {} }) => {
  return (
    <>
      <div onClick={onClickHandler}>
        <button className="w-[70px] h-[70px] border-2 rounded-md flex items-center justify-center border-[#EAEAEA] bg-[#FAFAFA] transition-all duration-1000">
          <div className="flex flex-row gap-1">
            <div className="w-[5px] h-[5px] bg-[#2F2F2F] rounded-full" />
            <div className="w-[5px] h-[5px] bg-[#2F2F2F] rounded-full" />
            <div className="w-[5px] h-[5px] bg-[#2F2F2F] rounded-full" />
          </div>
        </button>
      </div>
    </>
  );
};

export default MoreButton;
