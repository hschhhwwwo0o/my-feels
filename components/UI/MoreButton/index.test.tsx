import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import MoreButton from "./index";

test("Test MoreButton component", () => {
  render(
    <span data-testid="block">
      <MoreButton />
    </span>
  );
  const el: HTMLElement = screen.getByTestId("block");
  expect(el).toBeInTheDocument();
  expect(el).toMatchSnapshot();
});
