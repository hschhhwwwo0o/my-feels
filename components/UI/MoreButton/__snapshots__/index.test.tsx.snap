// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MoreButton component 1`] = `
<span
  data-testid="block"
>
  <div
    role="button"
  >
    <button
      class="w-[70px] h-[70px] border-2 rounded-md flex items-center justify-center border-[#EAEAEA] bg-[#FAFAFA] transition-all duration-1000"
    >
      <div
        class="flex flex-row gap-1"
      >
        <div
          class="w-[5px] h-[5px] bg-[#2F2F2F] rounded-full"
        />
        <div
          class="w-[5px] h-[5px] bg-[#2F2F2F] rounded-full"
        />
        <div
          class="w-[5px] h-[5px] bg-[#2F2F2F] rounded-full"
        />
      </div>
    </button>
  </div>
</span>
`;
